rpc/codec/client.go

```diff
@@ -108,9 +88,49 @@ func (c *clientCodec) ReadResponseBody(x interface{}) error {
 	return nil
 }
 
-// Close closes the underlying connection.
-func (c *clientCodec) Close() error {
-	return c.c.Close()
+func (c *clientCodec) writeRequest(id uint64, method string, request proto.Message) error {
+	// generate header
+	header := &c.reqHeader
+	*header = wire.RequestHeader{
+		Id:          id,
+		Method:      method,
+		Compression: wire.CompressionType_NONE,
+	}
+	if enableSnappy {
+		header.Compression = wire.CompressionType_SNAPPY
+	}
+
+	// marshal header
+	pbHeader, err := marshal(&c.reqBuf, header)
+	if err != nil {
+		return err
+	}
+
+	// send header (more)
+	if err := c.sendFrame(pbHeader); err != nil {
+		return err
+	}
+
+	// marshal request
+	pbRequest, err := marshal(&c.reqBuf, request)
+	if err != nil {
+		return err
+	}
+
+	// send body (end)
+	if enableSnappy {
```

Comment:
The threshold for when to switch from no compression to snappy compression is likely dependent on the connection between the client and server. It doesn't seem worth the effort to try to measure and account for that.